app/src/main/java/tech/relaycorp/gateway/domain/StoreParcel.kt

```diff
@@ -25,10 +25,15 @@ class StoreParcel
     suspend fun store(
         parcelStream: InputStream,
         recipientLocation: RecipientLocation
+    ) = store(parcelStream.readBytesAndClose(), recipientLocation)
+
+    @Throws(MalformedParcelException::class, InvalidParcelException::class)
+    suspend fun store(
+        parcelData: ByteArray,
+        recipientLocation: RecipientLocation
     ): StoredParcel {
-        val parcelBytes = parcelStream.readBytesAndClose()
         val parcel = try {
-            Parcel.deserialize(parcelBytes)
+            Parcel.deserialize(parcelData)
```

Comment:
(This comment applies to line 43, which is outside the diff)

We should enforce that `parcel.isRecipientAddressPrivate == true` when receiving parcels from the public endpoint. We shouldn't do that when the parcel originated in a private endpoint on the phone.

FYI: We have to pass the private gateway's certificate to `parcel.validate()` when the parcel is coming from the public gateway. We shouldn't do that when the parcel originated in a private endpoint on the phone. However, I forgot to implement this optional argument in `RAMFMessage.validate()`, so I've just added a TODO to #16 to fix this.

In other words, the two things above will depend on the value of `recipientLocation`.